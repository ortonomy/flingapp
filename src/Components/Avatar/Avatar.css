.Avatar {
  height: 100%;
  display: flex;
  flex-flow: row nowrap;
  align-items: center;
  justify-content: center;
  padding: 0 0.5em;
}

.Avatar div, img {
  padding: 0 0.5em;
}

.Avatar img {
  height: 3em;
  border-radius: 50%;
}

.Avatar .arrow-down {
  width: 0;
  height: 0;
  border-left: 5px solid transparent;
  border-right: 5px solid transparent;

  border-top: 5px solid black;
  padding: 0;
}
