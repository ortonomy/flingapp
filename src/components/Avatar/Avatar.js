import React, { Component } from 'react';
import styles from './Avatar.module.css';
import md5 from 'md5';

class Avatar extends Component {
  constructor(props) {
    super(props);
    this.getUserName = this.getUserName.bind(this);
    this.getEmailHash = this.getEmailHash.bind(this);
    this.getGravatarURL = this.getGravatarURL.bind(this);
  }

  getUserName() {
    const lastName = this.props.user.userLastName;
    const firstName = this.props.user.userFirstName;
    return `${firstName} ${lastName}`;
  }

  getEmailHash() {
    const email = this.props.user.userEmail.trim();
    return md5(email);
  }

  getGravatarURL(emailHash) {
    return `https://www.gravatar.com/avatar/${emailHash}`;
  }

  render() {
    this.getEmailHash();
    return(
      <div className={styles.Avatar}>
        <div className={styles.user}>{this.getUserName()}</div>
        <div className={styles.arrow}></div>
        <img className={styles.gravatar} src={this.getGravatarURL(this.getEmailHash())} alt="gravatar" />
      </div>
    )
  }
}

export default Avatar;
