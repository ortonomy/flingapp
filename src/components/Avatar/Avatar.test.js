import React from 'react';
import Avatar from '../Avatar';

const user = {
  userFirstName: "example",
  userLastName: "user",
  userEmail: "example@test.com"
}

it('renders without crashing', () => {
  const wrapper = shallow(<Avatar user={user} />);
  expect(wrapper).toMatchSnapshot();
});
