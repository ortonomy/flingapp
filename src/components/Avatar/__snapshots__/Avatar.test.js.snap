// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Avatar
    user={
        Object {
            "userEmail": "example@test.com",
            "userFirstName": "example",
            "userLastName": "user",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className={undefined}
>
          example user
</div>,
        <div
          className={undefined}
/>,
        <img
          alt="gravatar"
          className={undefined}
          src="https://www.gravatar.com/avatar/29e3f53ee49fae541ee0f48fb712c231"
/>,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "example user",
          "className": undefined,
        },
        "ref": null,
        "rendered": "example user",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "gravatar",
          "className": undefined,
          "src": "https://www.gravatar.com/avatar/29e3f53ee49fae541ee0f48fb712c231",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className={undefined}
>
            example user
</div>,
          <div
            className={undefined}
/>,
          <img
            alt="gravatar"
            className={undefined}
            src="https://www.gravatar.com/avatar/29e3f53ee49fae541ee0f48fb712c231"
/>,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "example user",
            "className": undefined,
          },
          "ref": null,
          "rendered": "example user",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "gravatar",
            "className": undefined,
            "src": "https://www.gravatar.com/avatar/29e3f53ee49fae541ee0f48fb712c231",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
